plugins/aws/ecr/ecrRepositoryEncrypted.js

```diff
@@ -0,0 +1,109 @@
+            desiredEncryptionLevelString: settings.ecr_repository_desired_encryption_level || this.settings.ecr_repository_desired_encryption_level.default
+        };
+
+        var desiredEncryptionLevel = helpers.ENCRYPTION_LEVELS.indexOf(config.desiredEncryptionLevelString);
+        var currentEncryptionLevel;
+
+        async.each(regions.ecr, function(region, rcb){
+            var describeRepositories = helpers.addSource(cache, source,
+                ['ecr', 'describeRepositories', region]);
+               
+            if (!describeRepositories) return rcb();
+            
+            if (describeRepositories.err || !describeRepositories.data) {
+                helpers.addResult(results, 3, `Unable to query ECR Repository: ${helpers.addError(describeRepositories)}`, region);
+                return rcb();
+            }
+            
+            if (!describeRepositories.data.length) {
+                helpers.addResult(results, 0, 'No ECR Repository  found', region);
+                return rcb();
+            }
+
+            var listKeys = helpers.addSource(cache, source,
+                ['kms', 'listKeys', region]);
+
+            if (!listKeys || listKeys.err || !listKeys.data) {
+                helpers.addResult(results, 3,
+                    `Unable to list KMS keys: ${helpers.addError(listKeys)}`, region);
+                return rcb();
+            }
+
+            for (let repository of describeRepositories.data) {
+                if (!repository.repositoryArn) continue;
+
+                let resource = repository.repositoryArn;
+
+                if ( repository.encryptionConfiguration &&
+                repository.encryptionConfiguration.encryptionType &&
+                    repository.encryptionConfiguration.encryptionType.toUpperCase() == 'AES256') {
+                    currentEncryptionLevel = 1;
+                } else if (repository.encryptionConfiguration && repository.encryptionConfiguration.kmsKey) {
+                    let kmsKey = repository.encryptionConfiguration.kmsKey;
+                    var keyId = kmsKey.split('/')[1] ? kmsKey.split('/')[1] : kmsKey;
+
+                    var describeKey = helpers.addSource(cache, source,
+                        ['kms', 'describeKey', region, keyId]);  
+
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, kmsKey);
+                        continue;
+                    }
+
+                    currentEncryptionLevel = helpers.getEncryptionLevel(describeKey.data.KeyMetadata, helpers.ENCRYPTION_LEVELS);
+                } else {
+                    helpers.addResult(results, 2,
+                        'ECR Repository does not have encryption enabled',
+                        region, resource);
```

Comment:
should not we have continue here? If this flow goes below currentEncryptionLevel value wont be set and result can be something unexpected.